engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingformula/CharRollingFormulaOperator.java

```diff
@@ -0,0 +1,205 @@
+            formulaCopy.initInputs(RowSetFactory.flat(1).toTracking(),
+                    Collections.singletonMap(PARAM_COLUMN_NAME, formulaInputSource));
+
+            formulaOutputSource = formulaCopy.getDataView();
+
+        }
+
+        @Override
+        public void close() {
+            super.close();
+            charWindowValues = null;
+        }
+
+        @Override
+        public void setValueChunks(@NotNull final Chunk<? extends Values>[] valueChunks) {
+            influencerValuesChunk = valueChunks[0].asCharChunk();
+        }
+
+        @Override
+        public void accumulateRolling(
+                @NotNull final RowSequence inputKeys,
+                @NotNull final Chunk<? extends Values>[] influencerValueChunkArr,
+                @Nullable final LongChunk<OrderedRowKeys> affectedPosChunk,
+                @Nullable final LongChunk<OrderedRowKeys> influencerPosChunk,
+                @NotNull final IntChunk<? extends Values> pushChunk,
+                @NotNull final IntChunk<? extends Values> popChunk,
+                final int len) {
+
+            setValueChunks(influencerValueChunkArr);
+            setPosChunks(affectedPosChunk, influencerPosChunk);
+
+            int pushIndex = 0;
+
+            // chunk processing
+            for (int ii = 0; ii < len; ii++) {
+                final int pushCount = pushChunk.get(ii);
+                final int popCount = popChunk.get(ii);
+
+                if (pushCount == NULL_INT) {
+                    writeNullToOutputChunk(ii);
+                    continue;
+                }
+
+                // pop for this row
+                if (popCount > 0) {
+                    pop(popCount);
+                }
+
+                // push for this row
+                if (pushCount > 0) {
+                    push(pushIndex, pushCount);
+                    pushIndex += pushCount;
+                }
+
+                // write the results to the output chunk
+                writeToOutputChunk(ii);
+            }
+
+            // chunk output to column
+            writeToOutputColumn(inputKeys);
+        }
+
+        @Override
+        public void push(int pos, int count) {
+            charWindowValues.ensureRemaining(count);
+
+            for (int ii = 0; ii < count; ii++) {
+                final char val = influencerValuesChunk.get(pos + ii);
+                charWindowValues.addUnsafe(val);
+            }
+        }
+
+        @Override
+        public void pop(int count) {
+            Assert.geq(charWindowValues.size(), "charWindowValues.size()", count);
+
+            for (int ii = 0; ii < count; ii++) {
+                charWindowValues.removeUnsafe();
+            }
+        }
+
+        @Override
+        public void writeToOutputChunk(int outIdx) {
+            // If not empty (even if completely full of null), run the formula over the window values.
+            outputSetter.accept(formulaOutputSource.get(0), outIdx);
```

Comment:
I suggested saving the `outputSetter` in this `Context` sub-class as an `IntConsumer` fetched via `getChunkSetter` passing in the output source so it can directly invoke `getChar` or `getInt` etc without needing to box.